### DISCUSSION TITLE
How to share server data to a client side script?

### DISCUSSION THREAD

**Discussion by PierBover** [freshframework/fresh#1481] [Q&A] [answered]
To put this out of the the way first, I know I can pass data to islands. This is not what I'm asking :)

I have an old school JS script in the static folder (eg: `static/something.js`) and need to share some data from the server to this script.

How I've done this in the past is serializing the data into JSON and putting it into a `<script>` tag like this:

```
<script id="json-data" type="application/json">${JSON.stringify(data)}</script>
```

But I'm wondering if Fresh has something out of the box?

**Comment by marvinhagemeister** [comment, marked as answer]
There is nothing wrong with writing data in a script tag. That's the way to do it. We could introduce some Fresh specific APIs, but then again I'm not sure if it would be "better" than passing a script tag or just the same thing a little different. The main thing the props serialization for islands add on top, is signals and bigint serialization.
